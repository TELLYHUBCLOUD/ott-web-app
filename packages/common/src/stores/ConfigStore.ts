import { ACCESS_MODEL, OTT_GLOBAL_PLAYER_ID } from '../constants';
import type { AccessModel, Config, IntegrationType } from '../../types/config';
import type { Settings } from '../../types/settings';

import { createStore } from './utils';

type ConfigState = {
  loaded: boolean;
  config: Config;
  accessModel: AccessModel;
  settings: Settings;
  integrationType: IntegrationType | null;
  isSandbox: boolean;
  clientId: string | null;
  offers: string[];
};

export const useConfigStore = createStore<ConfigState>('ConfigStore', () => ({
  loaded: false,
  config: {
    id: '',
    siteName: '',
    description: '',
    player: '',
    assets: {},
    content: [],
    menu: [],
    integrations: {
      cleeng: {
        useSandbox: true,
      },
      jwp: {
        clientId: null,
        assetId: null,
        useSandbox: true,
      },
    },
    styling: {
      footerText: '',
    },
  },
  settings: {
    additionalAllowedConfigSources: [],
    playerId: OTT_GLOBAL_PLAYER_ID,
  },
  accessModel: ACCESS_MODEL.SVOD,
  integrationType: null,
  isSandbox: true,
  clientId: null,
  offers: [],
}));
