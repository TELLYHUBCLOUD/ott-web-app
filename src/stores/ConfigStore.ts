import { createStore } from './utils';

import type { AccessModel, Config } from '#types/Config';
import { API_HOST, DEFAULT_PLAYER } from '#src/config';

export enum PersonalShelf {
  ContinueWatching = 'continue_watching',
  Favorites = 'favorites',
}

export const PersonalShelves = [PersonalShelf.Favorites, PersonalShelf.ContinueWatching];

type CleengData = {
  cleengId: string | null | undefined;
  cleengSandbox: boolean;
  monthlyOfferId: string;
  yearlyOfferId: string;
};

type ConfigState = {
  config: Config;
  accessModel: AccessModel;
  apiHost: string;
  getPlayer: () => string;
  getCleengData: () => CleengData;
};

export const useConfigStore = createStore<ConfigState>('ConfigStore', (_, get) => ({
  config: {
    id: '',
    siteName: '',
    description: '',
    player: '',
    assets: {},
    content: [],
    menu: [],
    integrations: {
      cleeng: {
        useSandbox: true,
      },
      inplayer: {
        clientId: null,
        assetId: null,
        useSandbox: true,
      },
    },
    styling: {
      footerText: '',
    },
  },
  accessModel: 'SVOD',
  apiHost: import.meta.env.APP_API_BASE_URL || API_HOST.prd,
  getPlayer: (): string => {
    const host = get().apiHost;

    // For dev api host we use only one testing id, for prod it can be also set in .env file
    return host === API_HOST.dev ? DEFAULT_PLAYER.dev : import.meta.env.APP_DEFAULT_PLAYER ?? DEFAULT_PLAYER.prd;
  },
  getCleengData: (): CleengData => {
    const cleeng = get().config?.integrations?.cleeng;

    const cleengId = cleeng?.id;
    const cleengSandbox = !!cleeng?.useSandbox;
    const monthlyOfferId = cleeng?.monthlyOffer || '';
    const yearlyOfferId = cleeng?.yearlyOffer || '';

    return { cleengId, cleengSandbox, monthlyOfferId, yearlyOfferId };
  },
}));
