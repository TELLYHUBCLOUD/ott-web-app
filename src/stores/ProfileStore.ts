import type { Profile } from '#types/account';
import { createStore } from '#src/utils/createStore';

type ProfileStore = {
  profile: Profile | null;
  selectingProfileAvatar: string | null;
};

export const useProfileStore = createStore<ProfileStore>('AccountStore', () => ({
  profile: null,
  selectingProfileAvatar: null,
}));
