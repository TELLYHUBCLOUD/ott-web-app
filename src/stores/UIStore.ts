import type { Location } from 'react-router-dom';

import { createStore } from '#src/utils/createStore';

type UIState = {
  searchQuery: string;
  searchActive: boolean;
  userMenuOpen: boolean;
  languageMenuOpen: boolean;
  preSearchPage?: Location;
};

export const useUIStore = createStore<UIState>('UIStore', () => ({
  searchQuery: '',
  searchActive: false,
  languageMenuOpen: false,
  userMenuOpen: false,
}));
