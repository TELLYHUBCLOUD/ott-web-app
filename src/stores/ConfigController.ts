import { useConfigStore } from './ConfigStore';

import { API_HOST } from '#src/config';
import { IS_PROD_MODE } from '#src/utils/common';

const envQueryKey = 'env';

export const initializeHost = (searchParams: URLSearchParams) => {
  const query = searchParams.get(envQueryKey);

  if (IS_PROD_MODE) {
    return;
  }

  if (query === 'dev' || query === 'prd') {
    useConfigStore.setState({ apiHost: API_HOST[query] });
    return;
  }

  useConfigStore.setState({ apiHost: API_HOST.prd });
};
