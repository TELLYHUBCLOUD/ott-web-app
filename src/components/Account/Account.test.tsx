import React from 'react';
import { waitFor } from '@testing-library/react';

import Account from './Account';

import { renderWithRouter } from '#test/testUtils';
import { useAccountStore } from '#src/stores/AccountStore';
import type { Consent } from '#types/account';
import customer from '#test/fixtures/customer.json';

describe('<Account>', () => {
  test('renders and matches snapshot', async () => {
    useAccountStore.setState({
      user: customer,
      publisherConsents: Array.of({ name: 'marketing', label: 'Receive Marketing Emails' } as Consent),
    });

    const { container } = renderWithRouter(<Account panelClassName={'panel-class'} panelHeaderClassName={'header-class'} canUpdateEmail={true} />);

    await waitFor(() => {
      const loadingOverlay = container.querySelector('[class*="loadingOverlay"]');
      return !loadingOverlay;
    });

    expect(container).toMatchSnapshot();
  });
});
