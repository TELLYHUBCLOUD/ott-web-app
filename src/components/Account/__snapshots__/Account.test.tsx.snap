// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Account> > renders and matches snapshot 1`] = `
<div>
  <div
    class="_loadingOverlay_eb61cb _inline_eb61cb"
  >
    <div
      class="_buffer_d122df"
    >
      <div />
      <div />
      <div />
      <div />
    </div>
  </div>
</div>
`;
